lib/Frontend/CompilerInvocation.cpp

```diff
@@ -968,6 +968,21 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
   for (const auto &Lib : Args.getAllArgValues(options::OPT_autolink_library))
     Opts.LinkLibraries.push_back(LinkLibrary(Lib, LibraryKind::Library));
 
+  if (const Arg *A = Args.getLastArg(OPT_type_info_dump_filter_EQ)) {
+    StringRef mode(A->getValue());
+    if (mode == "all")
+      Opts.TypeInfoFilter = IRGenOptions::TypeInfoDumpFilter::All;
+    else if (mode == "resilient")
+      Opts.TypeInfoFilter = IRGenOptions::TypeInfoDumpFilter::Resilient;
+    else if (mode == "fragile")
+      Opts.TypeInfoFilter = IRGenOptions::TypeInfoDumpFilter::Fragile;
+    else {
+      llvm::errs() << "Bad -type-info-dump-filter; expected one of "
+                   << "'all', 'fragile', 'resilient'\n";
```

Comment:
Please either make a proper diagnostic for this (like OPT_AssertConfig) or just use `llvm_unreachable`. The frontend doesn't need good diagnostics anyway; it's only for test authors.